### PR TITLE
refactor(vue/dialog): use v-model:open for two-way binding

### PR DESCRIPTION
**Pull request by iamdin**

### CODE REVIEW

File: packages/frameworks/vue/src/dialog/use-dialog.ts [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
   const getRootNode = useEnvironmentContext()
-  const reactiveContext = reactive(props)
 
   const [state, send] = useMachine(
     dialog.machine({
-      ...reactiveContext,
-      id: reactiveContext.id || useId().value,
+      ...context.value,
+      id: context.value.id || useId().value,
+      open: props.open ?? props.defaultOpen,
```
Comment (Omikorin) [inline packages/frameworks/vue/src/dialog/use-dialog.ts:26]: I found this in the Checkbox, should we change it to `checked: props.checked ?? props.defaultChecked` too?

image

File: packages/frameworks/vue/src/dialog/stories/dialog.story.vue [hunk 2]
```diff
@@ -9,7 +9,7 @@ const open = ref(false)
 <template>
   <Story title="Dialog">
     <Variant title="Basic">
-      <Dialog.Root>
+      <Dialog.Root default-open>
```
Comment (Omikorin) [inline packages/frameworks/vue/src/dialog/stories/dialog.story.vue:12]:
```suggestion
      <Dialog.Root>
```
To sync with other stories

**Review by Omikorin — APPROVED** [review]

**Comment by Omikorin** [comment]
Thank you very much!

File: packages/frameworks/vue/src/dialog/use-dialog.ts (hunk 1 above)
Comment (iamdin) [reply packages/frameworks/vue/src/dialog/use-dialog.ts:26]: yep, i missed it

Comment (Omikorin) [reply packages/frameworks/vue/src/dialog/use-dialog.ts:26]: I'll fix that

**Pull request merged**